internal/pkg/manifest/svc.go

```diff
@@ -365,57 +365,23 @@ type HTTPHealthCheckArgs struct {
 	GracePeriod        *time.Duration `yaml:"grace_period"`
 }
 
-func (h *HTTPHealthCheckArgs) isEmpty() bool {
+func (h *HTTPHealthCheckArgs) IsZero() bool {
 	return h.Path == nil && h.Port == nil && h.SuccessCodes == nil && h.HealthyThreshold == nil && h.UnhealthyThreshold == nil &&
 		h.Interval == nil && h.Timeout == nil && h.GracePeriod == nil
 }
 
 // HealthCheckArgsOrString is a custom type which supports unmarshaling yaml which
 // can either be of type string or type HealthCheckArgs.
 type HealthCheckArgsOrString struct {
-	HealthCheckPath *string
-	HealthCheckArgs HTTPHealthCheckArgs
-}
-
-// UnmarshalYAML overrides the default YAML unmarshaling logic for the HealthCheckArgsOrString
-// struct, allowing it to perform more complex unmarshaling behavior.
-// This method implements the yaml.Unmarshaler (v3) interface.
-func (hc *HealthCheckArgsOrString) UnmarshalYAML(value *yaml.Node) error {
-	if err := value.Decode(&hc.HealthCheckArgs); err != nil {
-		switch err.(type) {
-		case *yaml.TypeError:
-			break
-		default:
-			return err
-		}
-	}
-
-	if !hc.HealthCheckArgs.isEmpty() {
-		// Unmarshaled successfully to hc.HealthCheckArgs, reset hc.HealthCheckPath, and return.
-		hc.HealthCheckPath = nil
-		return nil
-	}
-
-	if err := value.Decode(&hc.HealthCheckPath); err != nil {
-		return errUnmarshalHealthCheckArgs
-	}
-	return nil
-}
-
-// IsEmpty returns true if there are no health check configuration set.
-func (hc *HealthCheckArgsOrString) IsEmpty() bool {
-	if hc.HealthCheckPath != nil {
-		return false
-	}
-	return hc.HealthCheckArgs.isEmpty()
+	Union[string, HTTPHealthCheckArgs]
```

Comment:
Instead of embeddding, can we just write the following?
```go 
type HealthCheckArgsOrString Union[string, HTTPHealthCheckArgs]
```


I'm thinking now with your configuration we can simplify a lot the names of our types, what do you think of the following?
```go
type AdvancedHTTPHealthCheckConfig struct {}
type HTTPHealthCheckConfig Union[string, AdvancedHTTPHealthCheckConfig]
```

Reply:
so i've been thinking about this option, but the main downside i've found is the inability to define functions on the new types. here's my synopsis on the different options (that i can think of!):

1. Embed in a struct (current)
```
type stringOrAdvanced struct {
  Union[string, advancedType]
}

func (s *stringOrAdvanced) Foo() string {
  return s.Union.Simple()
}
```
The main downside here is that you have to initialized the `stringOrAdvanced` struct every time you declare this type (see test files in this PR).

2. Create named type
```
type stringOrAdvanced *Union[string, advancedType]

// not allowed (compile err: invalid receiver type stringOrAdvanced (pointer or interface type) 
func (s *stringOrAdvanced) Hi() string {}
```
So the clear downside is that you can't define other functions, but it is nicer when declaring the type inline, though we'd probably have to change the `NewUnionXxx` functions to return a pointer instead to make that possible. We can't use `type stringOrAdvanced *Union[string, advancedType]` because then `stringOrAdvanced` no longer has an `UnmarshalYAML()` function on it...so it uses the default. I'm also not entirely sure if there is a potential for nil because of the pointer - it's kinda weird.

3. Type alias
```
type advancedType struct{}
type stringOrAdvanced = Union[string, advancedType]

// not allowed (compile err: cannot define new methods on instantiated type Union[string, advancedType])
func (s *stringOrAdvanced) Hi() {}
```
Same downside of the named type of not being able to declare a function, but I think avoids the potential nil oddity.

A helpful read: https://github.com/golang/proposal/blob/master/design/18130-type-alias.md

---
So those are the tradeoffs. If we wanted to do option 2 or 3, we could easily add functions onto their parent structs (i.e., `HealthCheckPath()`) to get around the function issue. Either way, to your point, we can remove the `ThisOrThat` naming convention from our types.

Reply:
Gotcha! okay so the maindownside for anyone else wondering is that if you write:
```go
type stringOrAdvanced Union[string, advancedType]
```
The method set of `Union` isn't embeded in `stringOrAdvanced`. Makes sense 👍 let's keep it embedded like you have!